E2E-ORS-Sample-For-Java/src/main/java/ObjectReplicationMonitor.java

```diff
@@ -0,0 +1,297 @@
+                            }
+                        }
+                    }
+                }
+
+                // Check if status has a new value and percentage, to avoid repeat printing the same status
+                if ((double) completed / (double) size != ratio && (int) (((double) completed / (double) size) * 100) != percentage) {
+                    ratio = (double) completed / (double) size;
+                    percentage = (int) (ratio * 100);
+
+                    // Output completion status as a percentage
+                    System.out.println("\nReplication completion is at " + percentage + "%...");
+                    System.out.println(completed + " out of " + size + " blobs have successfully replicated...");
+
+                    // Check for completion of all blobs, this will also cause end of while loop
+                    if (ratio == 1) {
+                        System.out.println("\nCompleted!");
+                        break;
+                    }
+                }
+            }
+        }
+        return completedEvents;
+    }
+
+
+    private static void checkContents(String sourceConnectionString, String sourceContainer, String destinationConnectionString, String destinationContainer, List<String> blobList) throws FileNotFoundException {
+        /*
+        This method is to check that the contents of a replicated blob transferred correctly
+         */
+
+        String[] connectionString = {sourceConnectionString, destinationConnectionString};
+        String[] containerName = {sourceContainer, destinationContainer};
+        for (int s = 0; s < connectionString.length; s++) {
+
+            // Creating blobserviceclient to access containerclient
+            BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionString[s]).buildClient();
+
+            // Get the container client to access blob client
+            BlobContainerClient blobContainerClient = blobServiceClient.getBlobContainerClient(containerName[s]);
+
+            // Print out contents of one blob to show successful replication
+            BlobClient blobClient = blobContainerClient.getBlobClient(blobList.get(0));
+
+            System.out.println("\nContent of " + blobList.get(0) + " in the " + containerName[s] + " container:");
+
+            // Create local file to store contents
+            File localFile = new File("src/main/java/replicated_blob_content.txt");
+
+            // Download blob content to file
+            blobClient.downloadToFile("src/main/java/replicated_blob_content.txt", true);
+
+            // Read content of file
+            Scanner fileReader = new Scanner(localFile);
+            while (fileReader.hasNextLine()) {
+                String content = fileReader.nextLine();
+                System.out.println(content);
+            }
+            fileReader.close();
+
+            // Delete content file-- clean up
+            localFile.deleteOnExit();
```

Comment:
You can also just download the blob to an OutputStream instead of saving it locally:
ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
blobClient.download(outputStream);
return new String(outputStream.toByteArray());